anko.go

```diff
@@ -17,75 +16,81 @@ import (
 	"github.com/mattn/anko/vm"
 
 	"github.com/daviddengcn/go-colortext"
-	"github.com/mattn/go-isatty"
 )
 
-const version = "0.0.2"
+const version = "0.0.3"
 
 var (
-	fs   = flag.NewFlagSet(os.Args[0], 1)
-	line = fs.String("e", "", "One line of program")
-	v    = fs.Bool("v", false, "Display version")
-
-	istty = isatty.IsTerminal(os.Stdout.Fd())
+	flagExecute *string
+	env         *vm.Env
 )
 
-func colortext(color ct.Color, bright bool, f func()) {
-	if istty {
-		ct.ChangeColor(color, bright, ct.None, false)
-		f()
-		ct.ResetColor()
-	} else {
-		f()
-	}
+func main() {
+	parseFlags()
+	setupEnv()
+	runNonInteractive()
+	runInteractive()
 }
 
-func main() {
-	fs.Parse(os.Args[1:])
-	if *v {
+func parseFlags() {
+	flagVersion := flag.Bool("v", false, "prints out the version and then exits")
+	flagExecute = flag.String("e", "", "execute the Anko code")
+	flag.Parse()
+
+	if *flagVersion {
 		fmt.Println(version)
 		os.Exit(0)
 	}
+}
 
-	var (
-		code      string
-		b         []byte
-		reader    *bufio.Reader
-		following bool
-		source    string
-	)
-
-	env := vm.NewEnv()
-	interactive := fs.NArg() == 0 && *line == ""
+func setupEnv() {
+	env = vm.NewEnv()
+	if len(*flagExecute) > 0 || flag.NArg() < 1 {
+		env.Define("args", flag.Args())
+	} else {
+		env.Define("args", flag.Args()[1:])
+	}
+	core.Import(env)
+	AddPackageColortext()
+	packages.DefineImport(env)
+}
 
-	env.Define("args", fs.Args())
+func runNonInteractive() {
+	if len(*flagExecute) < 1 && flag.NArg() < 1 {
+		return
+	}
 
-	if interactive {
-		reader = bufio.NewReader(os.Stdin)
-		source = "typein"
-		os.Args = append([]string{os.Args[0]}, fs.Args()...)
+	var source string
+	if len(*flagExecute) > 1 {
+		source = *flagExecute
 	} else {
-		if *line != "" {
-			b = []byte(*line)
-			source = "argument"
-		} else {
-			var err error
-			b, err = ioutil.ReadFile(fs.Arg(0))
-			if err != nil {
-				colortext(ct.Red, false, func() {
-					fmt.Fprintln(os.Stderr, err)
-				})
-				os.Exit(1)
-			}
-			env.Define("args", fs.Args()[1:])
-			source = filepath.Clean(fs.Arg(0))
+		sourceBytes, err := ioutil.ReadFile(flag.Arg(0))
+		if err != nil {
+			fmt.Println("ReadFile error:", err)
+			os.Exit(2)
 		}
-		os.Args = fs.Args()
+		source = string(sourceBytes)
 	}
 
-	core.Import(env)
-	AddPackageColortext()
-	packages.DefineImport(env)
+	_, err := env.Execute(source)
+	if err != nil {
+		fmt.Println("Execute error:", err)
+		os.Exit(4)
```

Comment:
let's return int value from runInteractive/runNonInteractive. and call os.Exit(r) in main

Reply:
Good idea